h/services/group.py

```diff
@@ -73,6 +73,32 @@ def create(self, name, authority, userid, description=None, type_='private'):
 
         return group
 
+    def create_open_group(self, name, authority, userid, hostnames, description=None):
+
+        def _get_or_create_groupscope(hostname):
+            try:
+                groupscope = self.session.query(GroupScope).filter_by(hostname=hostname).one()
+            except sa.orm.exc.NoResultFound:
+                groupscope = GroupScope(hostname=hostname)
+            return groupscope
+
+        scopes = [_get_or_create_groupscope(h) for h in hostnames]
```

Comment:
Before creating an open group you need to first create `GroupScope`s for each of its hostnames, if they don't exist already.